src/main/com/fulcrologic/rad/rendering/semantic_ui/field.cljc

```diff
@@ -36,11 +36,9 @@
              (when validation-message (str ent/nbsp "(" validation-message ")")))
            (div :.ui.input
              (input-factory (merge addl-props
-                              (cond->
-                                {:value    value
-                                 :onBlur   (fn [v] (form/input-blur! env qualified-key v))
-                                 :onChange (fn [v] (form/input-changed! env qualified-key v))}
-                                read-only? (assoc :readOnly "")))))
```

Comment:
This looks like a breaking change to me.  the readOnly flag is js is truthy, so an empty string should have been enough. could you double-check this?

Reply:
Thanks for reviewing the change Tony.

I was a little surprised as well, but this seems to be related to how React processes props. Try this out:
```html
<html>
  <head>
    <script crossorigin src="https://unpkg.com/react@17/umd/react.production.min.js"></script>
    <script crossorigin src="https://unpkg.com/react-dom@17/umd/react-dom.production.min.js"></script>
  </head>
  <body>
    Readonly = '' <div id="test1"></div>
    Readonly = 'readOnly' <div id="test2"></div>
    <script>
      ReactDOM.render(
        React.createElement('input', {readOnly: ''}, null),
        document.getElementById('test1')
      );
      ReactDOM.render(
        React.createElement('input', {readOnly: 'readOnly'}, null),
        document.getElementById('test2')
      );
    </script>
  </body>
</html>
```

When this is rendered, the readOnly property only comes through in the DOM in the second case:

<img width="301" alt="Read only" src="https://user-images.githubusercontent.com/24754123/114284197-30ad8b00-9a46-11eb-8106-5ad1ff312a82.png">

Reply:
ok, fair enough, but the intention was for *all* input fields to support readOnly, not just enumeations, and you removed it from the field.cljc file.

Reply:
Sorry, I should have been clearer in the commit. That was just a tidy up, the merge is still present, it's just the cond-> matching read-only? that has been removed.

Line 30 already had this:
```clj
addl-props         (if read-only? (assoc addl-props :readOnly "readonly") addl-props)]
```

So, there is no need to merge in :readOnly because it's already in addl-props. I only noticed this because it was changing the value for :readOnly from 'readOnly' to '' and so breaking the behaviour for read-only controls.